datalad/cmd.py

```diff
@@ -418,6 +416,72 @@ def run(self, cmd, protocol=None, stdin=None, cwd=None, env=None, **kwargs):
         results.pop('code', None)
         return results
 
+    @classmethod
+    def _check_if_new_proc(cls):
+        """Check if WitlessRunner is used under a new PID
+
+        Raises
+        ------
+        RuntimeError
+          If it is not a new proc and we already know that we need a new loop
+          in this pid
+        """
+        pid = os.getpid()
+        is_new_proc = cls._loop_pid is None or cls._loop_pid != pid
+        if is_new_proc:
+            # We need to check if we can run any command smoothly
+            cls._loop_pid = pid
+            cls._loop_need_new = None
+        elif cls._loop_need_new:
+            raise RuntimeError("we know we need a new loop")
```

Comment:
Not sure when this would happen, but maybe the error needs to be specific about which loop it is talking about and why this knowledge must be expressed as a runtimeerror.

Reply:
it is because of outside `try/except RuntimeError`.  This message will not be seen anywhere in outside code, so notion of a `loop` is clear as to me in this limited context.

Reply:
Well, that's the design I struggle with (although it is of course "valid if it works"). The function implements assumptions about how exactly and for what purpose it is called from within what context. Why not just return `True`/`False` and check for `_loop_need_new` outside of it?